src/main/java/decodes/tsdb/comprungui/CompRunGuiFrame.java

```diff
@@ -760,6 +780,69 @@ private void runButtonPressed()
 			return;
 		}
 
+		TsGroup compGroup = isSelectionValid(compVector);
+
+		if (compGroup == null)
+		{
+			Logger.instance().debug3("None of the selected computations uses a group.");
+		}
+		else if (compGroup.getGroupName() == null)
+		{
+			/* invalid group setup error already sent */
+			return;
+		}
+
+		progressBar.setStringPainted(true);
+		progressBar.setValue(0);
+		SwingWorker<?,?> runCompWorker = new SwingWorker<Vector<DbComputation>,Void>()
+		{
+
+			@Override
+			protected Vector<DbComputation> doInBackground() throws Exception
+			{
+				progressBar.setString("Processing Timeseries List.");
+				if (!buildComputationList(compVector, inputs, compGroup, (progress) -> setProgress(progress) ))
+				{
+					return null;
+				}
+				else
+				{
+					return compVector;
+				}
+			}
+
+			@Override
+			public void done()
+			{
+				try
+				{
+					setProgress(100);
+					Vector<DbComputation> theComps = this.get();
+					if (theComps != null)
+					{
+						runSelectedComps(theComps, originalLogger, traceLogger, inputs);
+					}
+					else
+					{
+						runButton.setEnabled(true);
+					}
+				}
+				catch (InterruptedException | ExecutionException ex)
+				{
+					runButton.setEnabled(true);
+					log.atError()
+					   .setCause(ex)
+					   .log("Unable to execute computations.");
+					showError(ex.getLocalizedMessage());
+				}
+			}
+		};
+		runCompWorker.addPropertyChangeListener(event -> updateProgress(event));
+		runCompWorker.execute();
+	}
+
+	private TsGroup isSelectionValid(Collection<DbComputation> compVector)
```

Comment:
isSelectionValid as a method name really 'feels' like it should return a Boolean.

Reply:
Hmm. yeah. That was my original intent but than couldn't return all the correct information.